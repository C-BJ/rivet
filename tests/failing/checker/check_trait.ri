trait ToStr {
    fn to_str(&self) str;
}

fn print(v: ToStr) void {
    _ = v.to_str();
}

extend str for ToStr {
    fn to_str(&self) str {
        return self.*;
    }
}

extend bool for ToStr {
    fn to_str(&self) str {
        return if (self.*) "true" else "false";
    }
}

fn main() void {
    print(""); // OK
    print(true); // OK
    print(1); // BAD
}
