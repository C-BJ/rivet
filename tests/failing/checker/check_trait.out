tests/failing/checker/check_trait.ri:24:11: error: type `i32` does not implement trait `ToStr`
     |
  24 |     print(1); // BAD
     |           ^
     = note: in argument `v` of function `print`
rivetc: error: could not compile package `check_trait`, aborting due to previous error.